x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/ltr/LearnToRankFeatureExtractorBuilder.java

```diff
@@ -8,15 +8,31 @@
 package org.elasticsearch.xpack.core.ml.inference.trainedmodel.ltr;
 
 import org.elasticsearch.common.io.stream.NamedWriteable;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.index.query.ParsedQuery;
+import org.elasticsearch.index.query.Rewriteable;
+import org.elasticsearch.index.query.SearchExecutionContext;
 import org.elasticsearch.xcontent.ParseField;
 import org.elasticsearch.xpack.core.ml.utils.NamedXContentObject;
 
-public interface LearnToRankFeatureExtractorBuilder extends NamedXContentObject, NamedWriteable {
+public interface LearnToRankFeatureExtractorBuilder
+    extends
+        NamedXContentObject,
+        NamedWriteable,
+        Rewriteable<LearnToRankFeatureExtractorBuilder> {
 
     ParseField FEATURE_NAME = new ParseField("feature_name");
 
     /**
      * @return The input feature that this extractor satisfies
      */
     String featureName();
+
+    /**
+     * @throws Exception If the extractor is invalid.
+     */
+    void validate() throws Exception;
```

Comment:
Its used in rewrite to verify that any queries stored as a feature are fully parseable. It may be that an older configuration is stored and a deprecated feature is removed. Then a stored query is no longer valid syntax and we should validate.